src/components/AddressAutocomplete.js

```diff
@@ -76,82 +76,116 @@ function AddressAutocomplete(props) {
   const [sessionToken, setSessionToken] = useState(null);
   const [controller, setController] = useState(null);
 
-  const fuse = new Fuse(addresses, fuseOptions);
-
-  const autocomplete = _.debounce((text, query) => {
-    const newController = new AbortController();
-    setController(newController);
-    const fuseResults = fuse.search(text, { limit: 2 });
-
-    if (country === 'gb') {
-      if (!postcode) {
-        axios
-          .get(
-            `https://api.postcodes.io/postcodes/${text.replace(
-              /\s/g,
-              '',
-            )}/autocomplete`,
-            { signal: newController.signal },
-          )
-          .then(response => {
-            if (
-              response.data.status === 200 &&
-              Array.isArray(response.data.result)
-            ) {
-              const normalizedPostcodes = response.data.result.map(
-                postcode => ({
-                  postcode: postcode,
-                  type: 'postcode',
-                }),
-              );
-              setResults(normalizedPostcodes);
-            }
-          })
-          .catch(error => {
-            console.log('AddressAutocomplete; _autocomplete', error);
-          });
-      } else {
-        setResults([
-          {
-            type: 'manual_address',
-            description: text,
-          },
-        ]);
-      }
-    } else {
-      axios
-        .get(
-          `https://maps.googleapis.com/maps/api/place/autocomplete/json?input=${encodeURIComponent(
-            text,
-          )}&${qs.stringify(query)}`,
-          { signal: newController.signal },
-        )
-        .then(response => {
-          const normalizedResults = fuseResults.map(fuseResult => ({
-            ...fuseResult.item,
-            type: 'fuse',
-          }));
-
-          const normalizedPredictions = response.data.predictions.map(
-            prediction => ({
-              ...prediction,
-              type: 'prediction',
-            }),
-          );
-
-          let results = normalizedResults.concat(normalizedPredictions);
-
-          if (normalizedResults.length > 0 && results.length > 5) {
-            results = results.slice(0, 5);
+  const fuse = useMemo(() => new Fuse(addresses, fuseOptions), [addresses]);
+
+  const autoCompleteDebounced = useMemo(
```

Comment:
I replaced `useCallback` with `useMemo` in https://github.com/coopcycle/coopcycle-app/pull/1954/commits/cb261ca1fa309200c14eb9d69f70bf48a2902e52 to fix this error:

![Screenshot 2025-02-24 at 12 07 13](https://github.com/user-attachments/assets/9887ab48-ac86-46a9-8928-6193968e9728)

Reply:
It's important to correctly list all parameters a hook depends on (`[country, fuse, postcode]);` part), otherwise it will use old values leading to a weird bugs